qemu/tests/migration_with_boot_pattern.py

```diff
@@ -0,0 +1,84 @@
+import aexpect
+import logging
+import time
+
+from autotest.client.shared import utils
+from autotest.client.shared import error
+
+
+@error.context_aware
+def run(test, params, env):
+    """
+    KVM migration test:
+    1) Start VM
+    2) Wait in background for the occurrence of the boot_pattern while
+       migrating the VM in a loop
+    3) Issue one more migration after the pattern is reached
+    4) Optionally send a line to serial port (to proceed booting) followed
+       by yet another migration
+    5) Login to the machine
+
+    :param test: kvm test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+    def wait_for_boot_pattern(boot_pattern, vm):
+        """
+        Waits till last_line_matches and survives vm-migration
+        """
+        patterns = [boot_pattern]
+        while True:
+            try:
+                if vm.serial_console.read_until_last_line_matches(patterns,
+                                                                  timeout=1):
+                    break
```

Comment:
I think here should be `return True`, otherwise line 71 would always raise exception.

Reply:
Oh True, and the condition there is actually bad...